pkg/registry/scripts/projects.js

```diff
@@ -271,6 +288,200 @@
+                } else {
+                    angular.forEach(subjects , function(s){
+                        if(s.kind === member.kind && s.name === member.name) {
+                            subjects.pop(s);
+                        }
+                    });         
+                }
+                return subjects;
+            }
+
+            function removeRoleData(member, kind, roleBinding) {
+                if (kind === "User")
+                    roleBinding.userNames = removeUsernames(member, roleBinding.userNames);
+                else
+                    roleBinding.groupNames = removeGroupNames(member, roleBinding.groupNames);
+                roleBinding.subjects = removeSubjects({
+                        kind: kind,
+                        name: member
+                    }, roleBinding.subjects);
+                return roleBinding;
+            }
+            function updateRoleData(member, kind, roleBinding) {
+                if (kind === "User")
+                    roleBinding.userNames = appendUsernames(member, roleBinding.userNames);
+                else
+                    roleBinding.groupNames = appendGroupNames(member, roleBinding.groupNames);
+                roleBinding.subjects = updateSubjects({
+                        kind: kind,
+                        name: member
+                    }, roleBinding.subjects);
+                return roleBinding;
+            }
+            function getRoleBinding(role, namespace) {
+                var rolebinding = null;
+                var default_policybinding = select().kind("PolicyBinding").namespace(namespace).name(":default");
+                if(default_policybinding.one()){
+                    var policybindings = default_policybinding.one();
+                    angular.forEach(policybindings.roleBindings, function(rb) {
+                        if(rb.name === role) {
+                            rolebinding = rb.roleBinding;
+                        }
+                    });
+                } 
+                return rolebinding;
+            }
+
+            function createRoleBinding(member_name, member_kind, role, namespace) {
+                var roleBindingDefault = {
+                    kind: "RoleBinding",
+                    apiVersion: "v1",
+                    metadata: {
+                        name: role,
+                        namespace: namespace,
+                        creationTimestamp: null,
+                    },
+                    userNames: [],
+                    groupNames: [],
+                    subjects: [],
+                    roleRef: {
+                        name: role
+                    }
+                };
+                var roleBinding = getRoleBinding(role, namespace);
+                if (roleBinding) {
+                    roleBinding = updateRoleData(member_name, member_kind, roleBinding);
+                    return methods.update(roleBinding, namespace);
```

Comment:
I believe we should be using PATCH here instead to avoid racing with multiple changes coming from multiple places. Is there a reason to use PUT instead?